snap-core/src/main/java/org/esa/snap/core/util/GeoUtils.java

```diff
@@ -0,0 +1,639 @@
+
+        return pathList.toArray(new GeneralPath[0]);
+    }
+
+    /**
+     * Creates a rectangular boundary expressed in pixel positions for the given source rectangle. If the source
+     * {@code rect} is 100 x 50 pixels and {@code step} is 10 the returned array will countain exactly 2 * 10
+     * + 2 * (5 - 2) = 26 pixel positions.
+     * <p>This method is used for an intermediate step when determining a raster boundary expressed in geographical
+     * co-ordinates.
+     *
+     * @param raster the raster
+     * @param rect   the source rectangle
+     * @param step   the mean distance from one pixel position to the other in the returned array
+     * @return the rectangular boundary
+     */
+    public static PixelPos[] createPixelBoundary(RasterDataNode raster, Rectangle rect, int step) {
+        final int width = raster.getRasterWidth();
+        final int height = raster.getRasterHeight();
+        return createPixelBoundary(width, height, rect, step);
+    }
+
+    /**
+     * Creates a rectangular boundary expressed in pixel positions for the given source rectangle. If the source
+     * {@code rect} is 100 x 50 pixels and {@code step} is 10 the returned array will countain exactly 2 * 10
+     * + 2 * (5 - 2) = 26 pixel positions.
+     * <p>This method is used for an intermediate step when determining a raster boundary expressed in geographical
+     * co-ordinates.
+     *
+     * @param rasterWidth  the raster width in pixels
+     * @param rasterHeight the raster height in pixels
+     * @param rect         the source rectangle
+     * @param step         the mean distance from one pixel position to the other in the returned array
+     * @return the rectangular boundary
+     */
+    public static PixelPos[] createPixelBoundary(int rasterWidth, int rasterHeight, Rectangle rect, int step) {
+        if (rect == null) {
+            rect = new Rectangle(0,
+                    0,
+                    rasterWidth,
+                    rasterHeight);
+        }
+        return createRectBoundary(rect, step);
+    }
+
+    /**
+     * Creates a rectangular boundary expressed in pixel positions for the given source rectangle. If the source
+     * {@code rect} is 100 x 50 pixels and {@code step} is 10 the returned array will countain exactly 2 * 10
+     * + 2 * (5 - 2) = 26 pixel positions.
+     * <p>This method is used for an intermediate step when determining a product boundary expressed in geographical
+     * co-ordinates.
+     * <p> This method delegates to {@link #createRectBoundary(java.awt.Rectangle, int, boolean) createRectBoundary(Rectangle, int, boolean)}
+     * and the additional boolean parameter {@code usePixelCenter} is {@code true}.
+     *
+     * @param rect the source rectangle
+     * @param step the mean distance from one pixel position to the other in the returned array
+     * @return the rectangular boundary
+     */
+    public static PixelPos[] createRectBoundary(Rectangle rect, int step) {
```

Comment:
Shouldn't this be named createPixelBoundary(...)?

Reply:
better: createPixelBoundaryFromRect()

what do you think?

Reply:
Agreed!